burr/core/application.py

```diff
@@ -2207,6 +2207,10 @@ def with_state_persister(
         """
         if on_every != "step":
             raise ValueError(f"on_every {on_every} not supported")
+        if hasattr(persister, 'initialize') and not persister.is_initialized():
```

Comment:
this is a good idea, but we need to add this to the persister interface and propagate it to the other persisters. To be backwards compatible with we would need to check for the existence of `is_initialized()` and only use it if available -- and therefore only error if it exists, else default to the old behavior of not checking.